RecoLocalCalo/EcalRecProducers/plugins/alpaka/EcalPhase2DigiToPortableProducer.cc

```diff
@@ -0,0 +1,76 @@
+#include "DataFormats/EcalDigi/interface/EcalDigiCollections.h"
+#include "DataFormats/EcalDigi/interface/EcalDigiPhase2HostCollection.h"
+#include "DataFormats/EcalDigi/interface/alpaka/EcalDigiPhase2DeviceCollection.h"
+#include "FWCore/Utilities/interface/EDGetToken.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/stream/EDProducer.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/Event.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/EventSetup.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/MakerMacros.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/EDPutToken.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+
+  class EcalPhase2DigiToPortableProducer : public stream::EDProducer<> {
+  public:
+    explicit EcalPhase2DigiToPortableProducer(edm::ParameterSet const &ps);
+    ~EcalPhase2DigiToPortableProducer() override = default;
+    static void fillDescriptions(edm::ConfigurationDescriptions &descriptions);
+
+    void produce(device::Event &event, device::EventSetup const &setup) override;
+
+  private:
+    const edm::EDGetTokenT<EBDigiCollectionPh2> inputDigiToken_;
+    const device::EDPutToken<EcalDigiPhase2DeviceCollection> outputDigiDevToken_;
```

Comment:
Just out of curiousity, does the `EcalDigiPhase2DeviceCollection` have other potential use than the input for `EcalUncalibRecHitPhase2WeightsProducerPortable`, or do you foresee other alternative modules that would produce `EcalDigiPhase2DeviceCollection` as well?

I'm asking only because if there is high certainty that this and `EcalUncalibRecHitPhase2WeightsProducerPortable` will always come together, and nothing else won't use `EcalDigiPhase2DeviceCollection` (or `EcalDigiPhase2HostCollection`), then these two modules could be merged into one module (that would lead to a somewhat simpler setup especially in the configuration). I'm not suggesting to change the setup in this PR, but only to give an aspect to think about.